### PR TITLE
fix(nav): pop() will unmount views within a modal

### PR DESCRIPTION
**Pull request by sean-perkins**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

When popping views with `ion-nav` used inside of a modal, the views are not unmounted. This results in broken experiences when trying to pop to the previous view or using other APIs dependent on pop, such as `popToRoot`. 

This was caused by inline modals, when the view is constructed `ion-nav` is disconnected from the view and re-attached. When disconnected, internal state was set to indicate that the Nav component was destroyed. This causes specific unmounting behavior not to fire. This flag was never reset to `true` when the component connected again.

Issue URL: #25637, #21831

## What is the new behavior?

- Views will correctly unmount when they are popped from an `ion-nav` inside an `ion-modal`.
- The internal destroyed state of `ion-nav` is reset when the component is connected to the DOM.
- Refactors internal methods for more explicit awareness to the behavior

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: core/src/components/nav/test/modal-navigation/nav.e2e.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    await page.goto(`/src/components/nav/test/modal-navigation`);
+    await openModal(page);
+  });
+
+  test('should render the root page', async ({ page }) => {
+    const pageOne = page.locator('page-one');
+    const pageOneHeading = page.locator('page-one h1');
+
+    await expect(pageOne).toBeVisible();
+    await expect(await pageOneHeading.textContent()).toBe('Page One');
```
Comment (liamdebeasi) [inline core/src/components/nav/test/modal-navigation/nav.e2e.ts:16]: You can probably use `toHaveText` instead

File: core/src/components/nav/test/modal-navigation/nav.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    await page.click('#go-back');
+
+    const pageTwo = page.locator('page-two');
+    const pageTwoHeading = page.locator('page-two h1');
+
+    // Verifies the leavingView was unmounted
+    await expect(pageThree).toHaveCount(0);
+    await expect(pageTwo).toBeVisible();
+    await expect(await pageTwoHeading.textContent()).toBe('Page Two');
```
Comment (liamdebeasi) [inline core/src/components/nav/test/modal-navigation/nav.e2e.ts:47]: You can probably use `toHaveText` instead

File: core/src/components/nav/test/modal-navigation/nav.e2e.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+      const pageOne = page.locator('page-one');
+      const pageOneHeading = page.locator('page-one h1');
+
+      // Verifies all views besides the root were unmounted
+      await expect(pageTwo).toHaveCount(0);
+      await expect(pageThree).toHaveCount(0);
+
+      await expect(pageOne).toBeVisible();
+      await expect(await pageOneHeading.textContent()).toBe('Page One');
```
Comment (liamdebeasi) [inline core/src/components/nav/test/modal-navigation/nav.e2e.ts:68]: You can probably use `toHaveText` instead

**Review by amandaejohnston — APPROVED** [review]
LGTM aside from Liam's comments.

**Review by liamdebeasi — APPROVED** [review]

**Pull request merged**